src/process_utility.c

```diff
@@ -752,6 +752,37 @@ rename_hypertable_constraint(Hypertable *ht, Oid chunk_relid, void *arg)
 	chunk_constraint_rename_hypertable_constraint(chunk->fd.id, stmt->subname, stmt->newname);
 }
 
+static void
+alter_hypertable_constraint(Hypertable *ht, Oid chunk_relid, void *arg)
+{
+	AlterTableCmd *cmd = (AlterTableCmd *) arg;
+	AlterTableCmd *chunk_cmd = copyObject(cmd);
```

Comment:
Is it strictly needed to do `copyObject` per chunk? I think the only thing you care about is that the original (hypertable) command is not modified (since it is passed through). But it doesn't matter if we modify chunk commands since they are internal. Therefore, you should be able to move the `copyObject` to the calling function. Still, might not be super-important. Still, always good to not eat more memory than necessary if we are dealing with thousands of chunks.

Reply:
The reason I did it this way is so that the cmd_con->conname is not modified on the original command, because it is used in the `chunk_constraint_get_name_from_hypertable_constraint` lookup

Reply:
Sorry, what I am saying is, copy once, not for every chunk.